### PR TITLE
implement index for users targets

### PR DESCRIPTION
**Pull request by lestairon**
## Description

Implement endpoint for the user to retrieve all created targets in json format

### CODE REVIEW

File: spec/requests/api/v1/targets_spec.rb [hunk 1]
```diff
@@ -103,4 +103,55 @@
...
+      end
+    end
+
+    context 'when user does not have targets' do
+      before { subject }
+
+      it 'returns an empty array' do
+        expect(parsed_response.size).to be_zero
+      end
+    end
```
Comment (Ricoch) [inline spec/requests/api/v1/targets_spec.rb:161]: WDYT of adding a "When the user is not signed in" case?

File: spec/requests/api/v1/targets_spec.rb [hunk 2]
```diff
@@ -103,4 +103,55 @@
...
+        expect(parsed_response).to all(have_key('longitude'))
+      end
+
+      it 'returns the target latitude' do
+        expect(parsed_response).to all(have_key('latitude'))
+      end
+
+      it 'returns the target radius' do
+        expect(parsed_response).to all(have_key('radius'))
+      end
```
Comment (Ricoch) [inline spec/requests/api/v1/targets_spec.rb:146]: we probably should also add a case for checking that the data is correct (it returns the different targets data, using match_array). Something like:
```ruby
    it 'returns all the targets' do
      expect(parsed_response.map(&:id)).to match_array(targets.map(&:id))
    end
```
might be enough

**Review by Ricoch — COMMENTED** [review]
PR looks great!

**Review by mrodriguez90 — APPROVED** [review]
Nice work!

File: spec/requests/api/v1/targets_spec.rb [hunk 3]
```diff
@@ -103,4 +103,71 @@
...
+        expect(targets_ids).to match_array(targets.map(&:id))
+      end
+    end
+
+    context 'when user does not have targets' do
+      before { subject }
+
+      it 'returns an empty array' do
+        expect(parsed_response.size).to be_zero
+      end
```
Comment (Ricoch) [inline spec/requests/api/v1/targets_spec.rb]: We could/should use the `be_empty` matcher here
```suggestion
      it 'returns an empty array' do
        expect(parsed_response).to be_empty
      end
```

**Review by Ricoch — APPROVED** [review]
Great job! One minor comment left, but it's not 100% necessary

**Pull request merged**